### PR TITLE
[stable/traefik] support for TLS settings on dashboard ingress

### PR DESCRIPTION
**Pull request by tobru**
This commit implements support for configuring TLS settings on the Traefik dashboard ingress object. Solves GitHub issue #6727

Signed-off-by: Tobias Brunner <tobias@tobru.ch>

#### What this PR does / why we need it:

Add support for TLS settings on the dashboard ingress object.

#### Which issue this PR fixes
- fixes ##6727

#### Special notes for your reviewer:

@krancour 
@emilevauge 
@dtomcej 
@ldez

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by tobru** [comment]
Fixed all linting issues, the PR is now ready for review.

File: stable/traefik/templates/dashboard-ingress.yaml [hunk 1]
```diff
@@ -27,4 +27,8 @@ spec:
       - backend:
           serviceName: {{ template "traefik.fullname" . }}-dashboard
           servicePort: 80
+  {{- if .Values.dashboard.ingress.tls }}
+  tls:
+{{ toYaml .Values.dashboard.ingress.tls | indent 4 }}
```
Comment (dtomcej) [inline stable/traefik/templates/dashboard-ingress.yaml:32]: Not a huge fan of this. It is not clear or user friendly. It is not obvious to new users what these mean, or what settings to use. 

I would suggest having the `ingress.tls` section of the values.yaml have elements that are parsed into this section here.

Thoughts?

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/dashboard-ingress.yaml (hunk 1 above)
Comment (tobru) [reply stable/traefik/templates/dashboard-ingress.yaml:32]: I don't see any benefit of mapping the possible sections of tls into the values. `toYaml` gives you the whole freedom of putting everything what's supported in tls into the values. And `toYaml` is widely used in official Helm charts. I see many other helm charts doing it this way, f.e. https://github.com/helm/charts/blob/master/stable/kubernetes-dashboard/templates/ingress.yaml#L43

**Comment by dtomcej** [comment]
@tobru can you bump the chart version so we can merge this?

Thanks 😊

**Comment by tobru** [comment]
@dtomcej version bumped :+1:

**Comment by cpanato** [comment]
@tobru can you please rebase?

**Comment by cpanato** [comment]
@tobru still a conflict in the chart.yaml

**Comment by cpanato** [comment]
can you give a final review @dtomcej thanks!

**Comment by tobru** [comment]
@cpanato Had to unpack some git fu, now it should be ok.

**Pull request merged**